03week/towersOfHanoi.js

```diff
@@ -20,17 +20,42 @@ function printStacks() {
 }
 
 function movePiece() {
-  // Your code here
-
+  let block = stacks.a.pop();
+  stacks.b.push(block);
 }
 
-function isLegal() {
+movePiece();
+console.log(stacks.a.length-1)
+
+function isLegal(startStack, endStack) {
+//   // let last = stacks.length-1
+//   let block = stacks.a.pop();
+//   let lasta = stacks.a.length-1
+//   let lastb = stacks.b.length-1
+//   let lastc = stacks.c.length-1
+//
+//   if ((block > stacks.b.length-1) || (block > stacks.c.length-1)) {
+//     return false;
+//   }
+//   if ((block < stacks[lastb]) || (block < stacks[lastc]) || (stacks[lastb] === 0) || (stacks[lastc] === 0)) {
+//     return true;
+//   }
+// }
+
+if (stacks[endStack].length === 0 || stacks[startStack].length - 1 < stacks[endStack].length - 1) {
+ return true;
+ } else {
+  return false;
+ }
+ };
   // Your code here
 
-}
-
 function checkForWin() {
-  // Your code here
+  if ((stacks.b.length || stacks.c.length) === 4) {
+    return true
+  } else {
+    return false;
+  }// Your code here
 
 }
 
```

Comment:
I think you forgot to put the function calls in the `towersOfHanoi()` functions to play the game